### PR TITLE
Add option to show icons on window hints

### PR DESCRIPTION
**Pull request by trishume**
I have implemented a feature that makes it easier to use window hints without ignoring hidden windows. I have added an option called `windowHintsShowIcons` that makes the background of window hints into the application's icon. This way you can tell what application a hint for a hidden window is for.

Example:
Screenshot

I also removed a bunch of Xcode user data files and changed the gitignore to ignore new ones.

### CODE REVIEW

**Comment by paddykontschak** [comment]
You might want to host the example somewhere else than Evernote, like [imgur](http://imgur.com/).

Evernote

**Comment by trishume** [comment]
Fixed

**Comment by trishume** [comment]
I have implemented an improvement that adds an additional option called windowHintsSpread that spreads out hints that are in the same place.

Here is what my example screenshot looks like with this option:
Screenshot

**Should I add this commit in the pull request?**

**Comment by jigish** [comment]
Yes, I like the windowHintsSpread idea. Please add that commit. Also, I think it may be useful to add a darker overlay over the icon because if the icon is lighter, the white text won't stand out enough (top-right case in your image). Or if you want to get really fancy, change the color of the text based on the icon :). This looks great though, thanks a lot!

**Comment by trishume** [comment]
There added the commit, although it shows up earlier in the pull request timeline.

The top right letter looks really hard to see in the mini screenshot but it is no trouble at all when it is full size.
The reason there is no dark background is that I had a bug that the icon would cause the background not to be drawn.
When I fixed it it looked really terrible with a dark overlay on it. So I added an if statement and turned it into a feature. :)

Although it might look okay if the dark square didn't take up the whole icon.

File: Slate/HintOperation.m [hunk 1]
```diff
@@ -34,10 +34,14 @@
...
 @implementation HintOperation
 
-@synthesize hints, windows, apps, hotkeyRefs, hideTimer, currentWindow, currentHint, hintCharacters;
+@synthesize hints, windows, apps, hotkeyRefs, frames, hideTimer, currentWindow, currentHint, hintCharacters;
 
 static const UInt32 ESC_HINT_ID = 10001;
 
+#define HINT_SPREAD_PADDING 20
+#define SPREAD_SEARCH_WIDTH 40
+#define SPREAD_SEARCH_HEIGHT 40
```
Comment (jigish) [inline Slate/HintOperation.m]: Sorry to be a pain, but can we make these values configurable as well? I like to make every possible value configurable :).

Comment (trishume) [inline Slate/HintOperation.m]: Sure, I'll do that soon

**Comment by trishume** [comment]
Fixed. Should I do anything else like make some options like hint spreading on by default. Should I pull your new changes from upstream or do you want to do the merge?

**Comment by jigish** [comment]
Looks great to me - Thanks a lot!

**Comment by trishume** [comment]
Thanks! Push a new version soon so I can stop using my dev copy?

**Comment by jigish** [comment]
Yah, I'll do it around 6pm Pacific

**Comment by jigish** [comment]
pushed :)

**Pull request merged**